src/api/route-services/gem2s.js

```diff
@@ -76,7 +79,15 @@ class Gem2sService {
       }, {});
     }
 
-    return taskParams;
+    const orderInvariantSampleIds = [...sampleIdsInOrder].sort();
+
+    const hashParams = {
+      sampleIds: orderInvariantSampleIds,
+      sampleNames: orderInvariantSampleIds.map((sampleId) => samples[sampleId].name),
+      metadata: taskParams.metadata,
+    };
+
+    return { taskParams, hashParams };
```

Comment:
I think we could add some unit testing to  this part at least to make sure that in the future we don't break this dependency on what causes it to run and what doesn't

Reply:
Written